src/server/transport/socketio.ts

```diff
@@ -17,76 +17,52 @@ import type {
 } from '../../master/master';
 import { getFilterPlayerView } from '../../master/filter-player-view';
 import type { Game, Server } from '../../types';
+import type { GenericPubSub } from './pubsub/generic-pub-sub';
+import { PubSubChannelIdNamespace } from './pubsub/generic-pub-sub';
+import type { IntermediateTransportData } from '../../master/master';
+import { InMemoryPubSub } from './pubsub/in-memory-pub-sub';
+import type { PubSubChannelId } from './pubsub/generic-pub-sub';
 
 const PING_TIMEOUT = 20 * 1e3;
 const PING_INTERVAL = 10 * 1e3;
 
+const emit = (socket: IOTypes.Socket, { type, args }: TransportData) => {
+  socket.emit(type, ...args);
+};
+
+function getPubSubChannelId(matchID: string): PubSubChannelId {
+  return { namespace: PubSubChannelIdNamespace.MATCH, value: matchID };
+}
+
```

Comment:
In fact, I think this is the only place that creates `PubSubChannelId` objects, right? And then passes them to the pub sub implementation. So this method could just do the string concatenation and we wouldn’t need the `globalChannelId` helper at all.
```suggestion
function getPubSubChannelId(matchID: string): string {
  return PubSubChannelIdNamespace.MATCH + '-' + matchID;
}

```

Reply:
Right, this had the advantaged of giving type safety and guaranteeing that the id wouldnt collide if they were used for other purposes beyond the match (maybe chat outside of a match in the future...). Generally over-using elementary types like string and etc to represent higher level concepts isnt advisable. But I followed your recommendation here.